shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -190,6 +190,16 @@ - (void)setupNotificationCenterObservers {
                object:nil];
 
   [center addObserver:self
+             selector:@selector(applicationDidEnterBackground:)
+                 name:UIApplicationDidEnterBackgroundNotification
```

Comment:
This changes the semantics of the states on iOS. Previously:
`willResignActive` --> `paused`
Now:
`willResignActive` --> `inactive`
`didEnterBackground` --> `paused`

Once we get this call, Apple documents the system gives the app ~5 seconds to do any last minute work it needs. We don't expose much in the way of kicking off iOS background tasks, delaying suspend, or getting at `[UIApplication backgroundTimeRemaining]` so developers can't rely on getting more than tiny things done here with any certainly. Though arguably they shouldn't be relying on being able to in the first place.